### ISSUE TITLE
New features - review and merge them

### ISSUE THREAD

**Issue by kirill-levitskiy** [elasticio/aws-s3-component#13]
There were implemented new reusable features compliant with OIH. They should be reviewed and merged into master branch.
PR: https://github.com/elasticio/aws-s3-component/pull/12

**Comment by shulkaolka** [comment]
**DoD**

**Integration application development prerequisites:**

- [x] Development team has access to the source with all requirements
- [ ] Meeting with Platform team representatives (for example Scrum Master) regarding E.IO platform changes (API etc.), which could potentially affect on Integration application development process
- [x] Access to the needed services with all needed permissions is granted
- [x] No open tasks that could block Integration application development process
- [x] Integration application architecture should be discussed and approved before starting coding

**Development:**

- [x] Code is written and formatted in accordance with the Styling guide
- [x]Code (incl. configs, test samples, environment variables etc.) MUST NOT contain any sensitive data like passwords, tokens, API keys etc.
- [x] Integration application logs does not contain any sensitive data (if needed due to requirements)
- [ ] Code is covered with unit tests
- [x] Code is covered with integration tests
- [x] Continuous deployment:
  - Publish script
  - Secrets
  - k8s descriptors
- [ ] Documentation:
  - Readme.md with sequence diagrams, overview, description and installation manual
  - Environment variables listing and description (env-vars.md)
  - Changelogs (CHANGELOG.md)

**After QA:**

- [ ] All development branches are reviewed and merged into master branch
- [ ] Sprint review preparation is finished.

**QA:**

- [ ] Test cases for the Integration application are ready at the TestRail
- [ ] Functional testing is finished
- [ ] Regression testing is finished (optional)
- [ ] Integration application is deployed to the required stage
- [ ] Smoke test is finished
- [ ] Release notes are done
